src/main/java/edu/cornell/kfs/fp/service/impl/AmazonWebServicesBillingServiceImpl.java

```diff
@@ -0,0 +1,446 @@
+import com.sun.jersey.api.client.config.DefaultClientConfig;
+
+import edu.cornell.kfs.fp.CuFPConstants;
+import edu.cornell.kfs.fp.document.CuDistributionOfIncomeAndExpenseDocument;
+import edu.cornell.kfs.fp.document.service.impl.CULegacyTravelServiceImpl;
+import edu.cornell.kfs.fp.service.AmazonWebServicesBillingService;
+import edu.cornell.kfs.fp.xmlObjects.AmazonAccountDetail;
+import edu.cornell.kfs.fp.xmlObjects.AmazonAccountDetailContainer;
+
+public class AmazonWebServicesBillingServiceImpl implements AmazonWebServicesBillingService, Serializable {
+
+    private static final long serialVersionUID = 7430710204404759511L;
+    private static final org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(AmazonWebServicesBillingServiceImpl.class);
+    
+    private static final String DEFAULT_BILLING_PERIOD_PARAMETER = "CURRENT";
+    private static final String INTERNAL_KFS_ACCOUNT_DESCRIPTION = "internal";
+    private static final String AUTHORIZATION_HEADER_NAME = "Authorization";
+    private static final String HTTP_METHOD_GET_NAME = "GET";
+    private static final String DI_ROUTE_ANNOTATION = "DI automatically created by AWS Billing Batch Job";
+    
+    private String awsURL;
+    private String awsToken;
+    private String billingPeriodParameterValue;
+    private String defaultDocumentDescription;
+    private String transactionObjectCode;
+    private String transactionFromAccountNumber;
+    private String helpNoteText;
+
+    protected ParameterService parameterService;
+    protected DocumentService documentService;
+    protected AccountService accountService;
+    protected BusinessObjectService businessObjectService;
+
+    @Override
+    public void generateDistributionOfIncomeDocumentsFromAWSService() {
+        LOG.info("generateDistributionOfIncomeDocumentsFromAWSService(): AWS URL: " + getAwsURL());
+        LOG.info("generateDistributionOfIncomeDocumentsFromAWSService(): Processing Year: " + findBillingYear() + "  Month: " + findBillingMonthNumber());
+        
+        String jsonResults = buildJSONoutput();
+        LOG.debug("JSON results: " + jsonResults);
+
+        List<AmazonAccountDetail> amazonAccountDetails = buildAmazonAcountListFromJson(jsonResults);
+        LOG.info("generateDistributionOfIncomeDocumentsFromAWSService() Number of amazon account details received: " + amazonAccountDetails.size());
+        
+        boolean valid = true;
+        
+        if (amazonAccountDetails.size() == 0) {
+            LOG.error("generateDistributionOfIncomeDocumentsFromAWSService() No amazon account detail objects parsed from the webservice.");
+            valid =  false;
+        }
+        
+        valid = valid && processAmazonAccountDetails(amazonAccountDetails);
+        
+        clearMemberVariables();
+        if (!valid) {
+            throw new RuntimeException("There was error processing AWS Billing.");
+        }
+    }
+    
+    private String buildJSONoutput() {
```

Comment:
I would probably capitalize O in output. Either buildJSONOutput or buildJsonOutput (I'm seeing more of this format lately I think).